### DISCUSSION TITLE
ALE project under MIT open source license?

### DISCUSSION THREAD

**Discussion by PennyAna** [joonaspaakko/audible-library-extractor#115] [General]
Hi, 
Your audible library extractor is precisely what I've been looking for!! Bonus: it's easy to use and the audiobook gallery is gorgeous. 
I'm a newer software engineer, so I hope you won't think that I'm being rude. I have a question about how I can use the ALE code. I'm building a personal project I want to use your code in. And I assume personal use is allowed without a problem. :) 
Eventually, I plan to turn my project into a marketable product. How do things work with using your ALE code at that point? Is it under the MIT open-source license? Should I make my own version when I sell the product? 
Thanks for answering my questions, I really appreciate any help you can give me! 
Nicole Lawrence (GitHub: PennyAna)

**Comment by joonaspaakko** [comment]
Hi. Going with what Github has to say about unlicensed code, which applies to this project:

> "You're under no obligation to choose a license. However, without a license, the default copyright laws apply, meaning that you retain all rights to your source code and no one may reproduce, distribute, or create derivative works from your work. If you're creating an open source project, we strongly encourage you to include an open source license."

I'm feeling hesitant to add any license right now since I have some concerns about it. There are cases where people have taken an open-source project and resold it as is, because all open-source licenses allow that as far as I know. I don't really care personally, but right now I'm in a place in my life, and with this project that I don't really need to add to my concerns. Even if they added cool stuff to it I'm afraid of the consequences of the hyena-like behavior I've seen in the Audible sub Reddit. It has taught me that a good chunk of Audible users are loud, are very quick to misunderstand things, and have no understanding of how digital purchases work in general. I'm pretty confident people would get at me if someone started selling a copy of ALE, because they think I sold out or whatever. 

> Honestly, after I get the next or possibly the next few versions out I might consider adding the MIT license but right now I'm not ready to really even think about it.

----

**That said, depending on what you want to do with it, I may permit you to use it / its code.**

> A few words of warning: 
>
> - Even if it had the MIT license, I can't guarantee its dependencies have any similarly permissive license. 
> - Commercial products that use web scraping even for the main functionality probably do still exist, but it's scary for a number of reasons. 
>     - Not just because the website can change and break everything in all previous versions (as more or less is the state of the extension right now) 
>     - ...But also because companies don't tend to like people scraping their websites and depending on how you look at it steal data and use up their bandwidth (which is what the extension does) and they may take action to prevent that. 
>         - ALE scapes like thousands of pages within like 1-10 minutes. I had a scare not too long ago, where they added some sort of rate limiting and it would basically kick your IP out of Audible for 15 minutes or so.
> - I'm no lawyer, but it is scraping data that is publicly available and some technically private data. Depending on how it's used, I'm not sure if they would like that. 
> - One of these days when I'm able to release the next version, it will be updated to Vue 3 (options API). I'm mostly saying this because I believe Vue 2 either already stopped getting security updates or it happens later this year.
>     - Also build tools were updated in general, because of changes in browser extension manifest v3.

**Reply by PennyAna** [reply]
Thank you so much for replying to my question! You've given me a lot to think about, especially about dependencies and legal concerns. Do you think Audible will ever create a public API? Part of my project will allow family members, especially children, to access a list of current family audiobooks that they can ask their Alexa device to play.

**Reply by joonaspaakko** [reply]
On one hand, I wouldn't say never... but I wouldn't hold my breath. It was I think 2021 when Amazon took down Goodreads API because it was old and they didn't want to spend time on the upkeep (or something like that), which might indicate they are not that into having any kinda public facing API. 

If I remember correctly, there's at least one Python project that provides [a wrapper for the private Audible API](https://pypi.org/project/audible/), which might be your best bet. Definitely better than scraping for data on the website. If you want another project that uses it as an example, I'm almost certain [Libation](https://github.com/rmcrackan/Libation) uses it.

**Comment by PennyAna** [comment]
Oooh, I'll look into these! I'm all for working smarter, not harder. :)
